Products/Poi/tests/base.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+from DateTime import DateTime
+from plone import api
+from plone.app.textfield import RichTextValue
+from plone.protect.interfaces import IDisableCSRFProtection
+from Products.Poi.adapters import IResponseContainer
+from Products.Poi.content.issue import next_issue_id
+from Products.Poi.testing import PRODUCTS_POI_INTEGRATION_TESTING
+from zope.event import notify
+from zope.interface import alsoProvides
+from zope.lifecycleevent import ObjectAddedEvent
+from zope.lifecycleevent import ObjectModifiedEvent
+
+import unittest
+
+
+class TestBase(unittest.TestCase):
+    layer = PRODUCTS_POI_INTEGRATION_TESTING
+
+    def rich_text(self, original):
+        return RichTextValue(
+            raw=original,
+            mimeType="text/html",
+            outputMimeType="text/html",
+            encoding="utf-8",
+        )
+
+    def addMember(self, username, fullname, email, roles, last_login_time):
+        # Taken from CMFPlone/tests/testMemberDataTool
+        self.portal.portal_membership.addMember(username, "secret", roles, [])
+        member = self.portal.portal_membership.getMemberById(username)
+        member.setMemberProperties(
+            {
+                "fullname": fullname,
+                "email": email,
+                "last_login_time": DateTime(last_login_time),
+            }
+        )
+
+    def createTracker(
+        self,
+        folder,
+        id,
+        title=u"",
+        description=u"",
+        helpText=u"",
+        availableAreas=[
+            {
+                "short_name": u"ui",
+                "title": u"User interface",
+                "description": u"User interface issues",
+            },
+            {
+                "short_name": u"functionality",
+                "title": u"Functionality",
+                "description": u"Issues with the basic functionality",
+            },
+            {
+                "short_name": u"process",
+                "title": u"Process",
+                "description": u"Issues relating to the development process itself",
+            },
+        ],
+        availableIssueTypes=[
+            {
+                "short_name": u"bug",
+                "title": u"Bug",
+                "description": u"Functionality bugs in the software",
+            },
+            {
+                "short_name": u"feature",
+                "title": u"Feature",
+                "description": u"Suggested features",
+            },
+            {
+                "short_name": u"patch",
+                "title": u"Patch",
+                "description": u"Patches to the software",
+            },
+        ],
+        availableSeverities=[u"Critical", u"Important", u"Medium", u"Low"],
```

Comment:
Please do not use a list as default argument. (Used multiple times in this file). Unexpected things can happen with that. See https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments
Better is to put in the function argument: `availableSeverities=None`, and then in the function code:

```
if availableSeverities is None:
    availableSeverities=[u"Critical", u"Important", u"Medium", u"Low"]
```

Reply:
you're right..fixed!